src/components/subscription-choices/subscription-choices.test.ts

```diff
@@ -0,0 +1,255 @@
+
+      // The form should now be visually hidden again
+      ({ formHeight, formWidth } = await form.evaluate(getFormDimensions));
+      expect(formHeight).toBeLessThanOrEqual(1);
+      expect(formWidth).toBeLessThanOrEqual(1);
+
+      // Navigate forward past the Submit to activate the form hide timeout
+      await page.keyboard.press('Tab'); // Email input
+      await page.keyboard.press('Tab'); // Submit button
+      await page.keyboard.press('Tab'); // Out of the form
+
+      // Confirm the form is still "active" (not visually hidden)
+      ({ formHeight, formWidth } = await form.evaluate(getFormDimensions));
+      expect(formHeight).toBeGreaterThan(1);
+      expect(formWidth).toBeGreaterThan(1);
+
+      // Navigate back quickly to confirm timeout getting cancelled
+      await page.keyboard.down('Shift'); // Navigate backwards
+      await page.keyboard.press('Tab'); // Submit button
+      await page.keyboard.up('Shift'); // Release Shift key
+
+      // Confirm the form is still "active" (not visually hidden)
+      ({ formHeight, formWidth } = await form.evaluate(getFormDimensions));
+      expect(formHeight).toBeGreaterThan(1);
+      expect(formWidth).toBeGreaterThan(1);
+
+      await page.keyboard.press('Tab'); // Out of the form
+
+      // Confirm the form is still "active" (not visually hidden)
+      ({ formHeight, formWidth } = await form.evaluate(getFormDimensions));
+      expect(formHeight).toBeGreaterThan(1);
+      expect(formWidth).toBeGreaterThan(1);
+
+      // After a timeout, the form eventually visually hides
+      await waitFor(
+        async () => {
+          ({ formHeight, formWidth } = await form.evaluate(getFormDimensions));
+          expect(formHeight).toBeLessThanOrEqual(1);
+          expect(formWidth).toBeLessThanOrEqual(1);
+        },
+        {
+          timeout: 2000,
+          interval: 1000,
+        }
+      );
+    })
+  );
+
+  test(
+    'should be customizable',
+    withBrowser(async ({ utils, screen }) => {
+      // Set up CSS
+      await loadGlobalCSS(utils);
+      await utils.loadCSS('./subscription-choices.scss');
+
+      // No customization
+      await utils.injectHTML(await componentMarkup({ form_id: 'test' }));
+
+      // Confirm default heading tag
+      await screen.getByRole('heading', {
+        name: 'Never miss an article!',
+        level: 2,
+      });
+      await screen.getByRole('heading', {
+        name: 'Get Weekly Digests',
+        level: 2,
+      });
+
+      // Confirm default form values
+      let form = await screen.getByRole('form');
+      let { formAction, emailInputPlaceHolder } = await form.evaluate(
+        (formEl) => ({
+          formAction: formEl.getAttribute('action'),
+          emailInputPlaceHolder: formEl.querySelector('input')?.placeholder,
+        })
+      );
+      expect(formAction).toBe(
+        'https://cloudfour.us13.list-manage.com/subscribe/post?u=ce064f42c86a5982dd218d4de&amp;id=7e505a6a67'
+      );
```

Comment:
Can we use [`.toHaveAttribute`](https://github.com/testing-library/jest-dom#tohaveattribute) here? (for placeholder too)?

Reply:
https://github.com/cloudfour/cloudfour.com-patterns/pull/1679/commits/9134022b79a690bc2a1bbda7a0933b522505be5d